### PR TITLE
Binheap

### PR DESCRIPTION
**Pull request by aepaysinger**
this is just for the min heap. i see that i have some comments i need to remove as well. If everything else looks good i will work on the max heap. and then make it so someone can pick min or max

### CODE REVIEW

File: min_binheap.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# _for private methods
```
Comment (W-Ely) [inline min_binheap.py]: Remove comment, can add it to your nots somewhere if it's helpful

File: min_binheap.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    def _has_left_child(self, index):
+        return self._find_left_child_index(index) < len(self.storage)
+
+    def _has_right_child(self, index):
+        return self._find_right_child_index(index) < len(self.storage)
+
+    def _parent_location(self, index):
+        if index == 0:
+            raise ValueError("Root Element")
```
Comment (W-Ely) [inline min_binheap.py]: During normal usage of this Heap, is it possible for this exception to happen?

File: min_binheap.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        return self.storage[self._find_parent_index(index)]
+
+    def _left_child_location(self, index):
+        if self._find_left_child_index(index) <= len(self.storage) and index % 2 != 0:
+            raise ValueError("Has no left child")
```
Comment (W-Ely) [inline min_binheap.py]: During normal usage of this Heap, is it possible for this exception to happen?

File: min_binheap.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        return self.storage[self._find_left_child_index(index)]
+
+    def _right_child_location(self, index):
+        if self._find_right_child_index(index) >= len(self.storage):
+            raise ValueError("Has no right child")
```
Comment (W-Ely) [inline min_binheap.py]: During normal usage of this Heap, is it possible for this exception to happen?

File: min_binheap.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        ):
+            self._swap(self._find_parent_index(index), index)
+            index = self._find_parent_index(index)
+
+    def pop(self):
+        if self.storage == []:
+            raise ValueError("Empty heap")
+        else:
+            self.storage[0] = self.storage[-1]
+            self.storage = self.storage[0:-1]
```
Comment (W-Ely) [inline min_binheap.py]: Is the purpose of this line `self.storage = self.storage[0:-1]` just to remove the last element?  Doing a slice creates a new list, what is a better way to remove the last item from a list?

File: min_binheap.py (hunk 2 above)
Comment (aepaysinger) [reply min_binheap.py]: i guess not. should i remove them all. just extra that i don't need

Comment (W-Ely) [reply min_binheap.py]: They should be removed. Probably they were helpful when figuring out how to do it, but they are no longer needed.

**Review by W-Ely — APPROVED** [review]

**Pull request merged**